### PR TITLE
Fix attributes on copy paste while QGIS stays responsive

### PR DESCRIPTION
**Pull request by signedav**
By using `show()` instead of `exec()` the `QgsFixAttributeDialog` to fix features on Copy/Paste if the constraints are not fulfilled, the user is able to do stuff meanwhile. Like e.g. create a feature on the parent layer by using the + in the Relation Reference Widget.

Since this fixes a behavior that is not nice I like to backport it as well.

### CODE REVIEW

**Comment by andreasneumann** [comment]
That's an important fix! People have been tracing existing geometries instead of copy/paste because of this behaviour.

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -10214,26 +10214,43 @@ void QgisApp::pasteFromClipboard( QgsMapLayer *destinationLayer )
...
-      QgsFixAttributeDialog *dialog = new QgsFixAttributeDialog( pasteVectorLayer, invalidFeatures, this );
-      int feedback = dialog->exec();
+      QgsAttributeEditorContext context( createAttributeEditorContext() );
+      context.setAllowCustomUi( false );
+      context.setFormMode( QgsAttributeEditorContext::StandaloneDialog );
 
-      switch ( feedback )
+      QgsFixAttributeDialog *dialog = new QgsFixAttributeDialog( pasteVectorLayer, invalidFeatures, this, context );
+
+      connect( dialog, &QgsFixAttributeDialog::finished, this, [ = ]( int feedback )
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp:10223]: This will leak if the dialog is cancelled

Comment (m-kuhn) [reply src/app/qgisapp.cpp:10223]: Out of interest.

The [QDialog docs](https://doc.qt.io/qt-5/qdialog.html) mention

> e.g. an OK button to the accept() slot and a Cancel button to the reject() slot.
> [...]
> If the user presses the Esc key in a dialog, QDialog::reject() will be called.

and

> QDialog::finished(int result) : This signal is emitted when the dialog's result code has been set, either by the user or by calling done(), accept(), or reject().

... but I cannot exclude that there's a way to get out of the dialog without deleting it. What would be the proper solution?

Comment (nyalldawson) [reply src/app/qgisapp.cpp:10223]: So does pressing cancel in this dialog still result in the paste happening?

Comment (m-kuhn) [reply src/app/qgisapp.cpp:10223]: I assumed that's catched in the switch below, but @signedav is in the better position to answer.

Comment (signedav) [reply src/app/qgisapp.cpp:10223]: On canceling the dialog, `features` stays empty. Means we call `pasteFeatures` with 0 features and raise there the warning message that no features have  been copied. We could catch this case earlier, but it means to have a duplicate code for the message.

Comment (nyalldawson) [reply src/app/qgisapp.cpp:10223]: Ok, fair enough!

**Review by m-kuhn — APPROVED** [review]

File: src/app/qgisapp.cpp (hunk 1 above)
Comment (signedav) [reply src/app/qgisapp.cpp:10223]: Thanks @nyalldawson @m-kuhn

**Comment by andreasneumann** [comment]
Any chance we can backport this to 3.10 as well? It is a very annoying problem for us.

**Pull request merged**